lib/ramble/ramble/graphs.py

```diff
@@ -0,0 +1,424 @@
+                            deps.append(dep_node)
+
+                        exec_node = self.node_definitions[builtin]
+                        super().update_graph(exec_node, deps)
+
+    def inject_executable(self, exec_name, injection_order, relative):
+        """Inject an executable into the graph
+
+        Args:
+            exec_name (str): Name of executable to inject
+            injection_order (str): Order for injection. Can be 'before' or 'after'
+            relative (str): Name of executable to inject relative to. Can be
+                            None to inject relative to the whole set of executables.
+        """
+        # Order can be 'before' or 'after.
+        # If `relative_to` is not set, then before adds to be the beginning of the list
+        #                  and after (default) adds to the end of the list
+        # If `relative_to` IS set, then before adds before the first instance of
+        #                  the executable in the list
+        #                  and after (default) adds after the last instance of the
+        #                  executable in the list
+        # If `relative_to` is set, and the executable name is not found, raise a fatal error.
+
+        exec_node = self.node_definitions[exec_name]
+        cur_exec_order = []
+        for node in self.walk():
+            cur_exec_order.append(node)
+
+        exp_name = self._obj_inst.expander.experiment_namespace
+        order = self.supported_injection_orders.after
+        if injection_order is not None:
+            if not hasattr(self.supported_injection_orders, injection_order):
+                logger.die('In experiment '
+                           f'"{exp_name}" '
+                           f'injection order of executable "{exec_name}" is set to an '
+                           f'invalid value of "{injection_order}".\n'
+                           f'Valid values are {self.supported_injection_orders}.')
+            order = getattr(self.supported_injection_orders, injection_order)
+
+        if exec_name not in self.node_definitions:
+            logger.die('In experiment '
+                       f'"{exp_name}" '
+                       f'attempting to inject a non existing executable "{exec_name}".')
+
+        if relative is not None:
+            relative_error = False
+            if relative not in self.node_definitions:
+                relative_error = True
+
+            relative_node = self.node_definitions[relative]
+            if relative_node not in cur_exec_order:
+                relative_error = True
+
+            if relative_error:
+                logger.die('In experiment '
+                           f'"{exp_name}" '
+                           f'attempting to inject executable "{exec_name}" '
+                           f'relative to a non existing executable "{relative}".')
+
+        if relative is not None and relative not in self.node_definitions:
```

Comment:
I actually think this is dead code. And looking at the assignment below it, it should always error since we key off of relative in node_definitions, but we already checked that it wasn't in node_definitions. (And this exact error case is already handled above).

So, I'll just remove it.